pkg/blobinfocache/sqlite/sqlite.go

```diff
@@ -467,6 +479,19 @@ func (sqc *cache) appendReplacementCandidates(candidates []prioritize.CandidateW
 	if err := rows.Err(); err != nil {
 		return nil, fmt.Errorf("iterating through locations: %w", err)
 	}
+
+	if len(res) == 0 && requireCompressionInfo && digest.String() != "" {
```

Comment:
The other implementations are conditioned on something called `v2Output`. Why is it not necessary here?

Reply:
Done.